sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -397,6 +400,8 @@ def fit(self, X, y, sample_weight=None):
         # train/val split).
         if not (self.warm_start and self._is_fitted()):
             self._random_seed = rng.randint(np.iinfo(np.uint32).max, dtype="u8")
+            self._random_seed2 = rng.randint(np.iinfo(np.uint32).max, dtype="u8")
+        rng2 = np.random.default_rng(self._random_seed2)
```

Comment:
IIUC, this means that we still have the same `rng2` for 2 consecutive calls to `fit()` when doing warm-start? i.e. the growers of the second call to `fit()` will have the same RNG as the growers from the first call to `fit()`.

Maybe it's not a big deal. Otherwise we could do something like

```suggestion
            feature_subsample_seed =  rng.randint(np.iinfo(np.uint32).max, dtype="u8")
            self._feature_subsample_rng = np.random.default_rng(feature_subsample_seed)
```

and pass `self._feature_subsample_rng` to the growers.